crates/ibc/src/core/ics23_commitment/merkle.rs

```diff
@@ -44,8 +44,10 @@ impl From<RawMerkleProof> for MerkleProof {
             .into_iter()
             .map(|p| {
                 let mut encoded = Vec::new();
-                prost::Message::encode(&p, &mut encoded).unwrap();
-                prost::Message::decode(&*encoded).unwrap()
+                prost::Message::encode(&p, &mut encoded)
+                    .expect("Never fails to encode because of the same type");
+                prost::Message::decode(&*encoded)
+                    .expect("Never fails to decode because of the same type")
```

Comment:
I believe this is just checking if the data is correct, using encoding and decoding.

Reply:
I'll double check to make sure it's really safe to remove

Reply:
Confirmed, we had this because the old ibc-proto-rs used its own ics23.rs. We don't need this encoding/decoding anymore since [this](https://github.com/cosmos/ibc-proto-rs/commit/0caffd87dac4dee9f0837a4d1b422b2d1b128ee0).